MoltenVK/MoltenVK/GPUObjects/MVKQueue.mm

```diff
@@ -292,8 +296,27 @@
 		}];
 	}
 
-	// Use temp var because callback may destroy this instance before this function ends.
+	// Use temp vars because callback may destroy this instance before this function ends.
+	MVKDevice* device = _queue->getDevice();
 	id<MTLCommandBuffer> mtlCmdBuff = _activeMTLCommandBuffer;
+	// If command buffer execution fails, log it, and mark the device lost.
+	[mtlCmdBuff addCompletedHandler: ^(id<MTLCommandBuffer> mtlCmdBuff) {
+		if (mtlCmdBuff.status == MTLCommandBufferStatusError) {
+			device->reportError(device->markLost(), "Command buffer %p \"%s\" execution failed (code %li): %s", mtlCmdBuff, mtlCmdBuff.label ? mtlCmdBuff.label.UTF8String : "", mtlCmdBuff.error.code, mtlCmdBuff.error.localizedDescription.UTF8String);
+			// Some errors indicate we lost the physical device as well.
+			switch (mtlCmdBuff.error.code) {
+				case MTLCommandBufferErrorBlacklisted:
+				// XXX This may also be used for command buffers executed in the background without the right entitlement.
+				case MTLCommandBufferErrorNotPermitted:
+#if MVK_MACOS
+				case MTLCommandBufferErrorDeviceRemoved:
+#endif
+					device->getPhysicalDevice()->setConfigurationResult(VK_ERROR_DEVICE_LOST);
```

Comment:
Setting configuration of `MVKPhysicalDevice` here. Most tests above seem to be testing against `MVKDevice`. The only place I can see where the config result of `MVKDevice` is set is in the constructor. Should we not be setting that value dynamically too? Perhaps I'm not seeing it...or missing something in the logic?

Reply:
`MVKDevice::markLost()` sets the configuration result to `VK_ERROR_DEVICE_LOST`.

Reply:
D'oh. Right. I don't know how I missed that.